clock/test_clock_test.go

```diff
@@ -61,3 +63,28 @@ func TestTickAfter(t *testing.T) {
 	tickOrTimeOut(ticker2, true)
 	tickOrTimeOut(ticker3, false)
 }
+
+// TestTickSignal tests that TickAfter signals registration allowing
+// safe time advancement.
+func TestTickSignal(t *testing.T) {
+	ch := make(chan struct{})
+	c := NewTestClockWithTickSignal(testTime, ch)
+
+	var wg sync.WaitGroup
+	wg.Add(1)
+
+	go func() {
+		defer wg.Done()
+
+		select {
+		case <-c.TickAfter(time.Second):
+
+		case <-time.After(time.Second):
+			t.Errorf("timeout")
```

Comment:
This call isn't allowed in a goroutine afaik

Reply:
TIL. This makes the test a bit more complicated to read..